### PR TITLE
 (GH-121) Support message in multiple format on issues 

### LINKED ISSUE #121: Support reading messages in different formats
Currently message format can be defined while reading issues. This means that if issues are processed for different purposes (e.g. written to a pull request and a report created) then format might not fit for all use cases. It might be better to allow reading different (or all) message formats and let the consumer decide which formats is used.

### PR DESCRIPTION
**Pull request by pascalberger**
Supports messages in multiple format on issues. Instead of deciding in which format the message should be provided by the issue provider, `IIssue` contains now the message in all formats and the consumer can decide which of the formats fits the best.

Fixes #121

### CODE REVIEW

File: src/Cake.Issues/Serialization/SerializableIssueV2Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        /// <summary>
+        /// Converts a <see cref="SerializableIssueV2"/> to an <see cref="Issue"/>.
+        /// </summary>
+        /// <param name="serializableIssue">Issue which should be converted.</param>
+        /// <returns>Converted issue.</returns>
+        internal static Issue ToIssue(this SerializableIssueV2 serializableIssue)
+        {
+#pragma warning disable SA1123 // Do not place regions within elements
+            #region DupFinder Exclusion
```
Comment (christianbumann) [inline src/Cake.Issues/Serialization/SerializableIssueV2Extensions.cs]: Why a region in such a small class?

Comment (pascalberger) [reply src/Cake.Issues/Serialization/SerializableIssueV2Extensions.cs]: @christianbumann This is the only way to exclude part of a file from dup finder, which I'm aware of (would be happily use another one if available)

File: src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs [hunk 2]
```diff
@@ -574,13 +666,67 @@ public void Should_Throw_If_FilePath_Is_Null()
...
+                    // When
+                    issue.SerializeToJsonFile(filePath);
+                    var result = filePath.DeserializeToIssue();
+
+                    // Then
+                    result.MessageText.ShouldBe(messageText);
+                }
+                finally
+                {
+                    System.IO.File.Delete(filePath.FullPath);
```
[the remark this answers is not in the document]
Comment (pascalberger) [reply src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs]: Good point. I'll do this in a subsequent PR over the whole codebase.

File: src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs [hunk 3]
```diff
@@ -574,13 +666,67 @@ public void Should_Throw_If_FilePath_Is_Null()
...
+                    // When
+                    issue.SerializeToJsonFile(filePath);
+                    var result = filePath.DeserializeToIssue();
+
+                    // Then
+                    result.MessageMarkdown.ShouldBe(messageMarkdown);
+                }
+                finally
+                {
+                    System.IO.File.Delete(filePath.FullPath);
```
[the remark this answers is not in the document]
Comment (pascalberger) [reply src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs]: Good point. I'll do this in a subsequent PR over the whole codebase.

File: src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs [hunk 4]
```diff
@@ -899,19 +1045,93 @@ public void Should_Throw_If_FilePath_Is_Null()
...
+                    var result = filePath.DeserializeToIssues();
+
+                    // Then
+                    result.Count().ShouldBe(2);
+                    result.First().MessageText.ShouldBe(messageText1);
+                    result.Last().MessageText.ShouldBe(messageText2);
+                }
+                finally
+                {
+                    System.IO.File.Delete(filePath.FullPath);
```
[the remark this answers is not in the document]
Comment (pascalberger) [reply src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs]: Good point. I'll do this in a subsequent PR over the whole codebase.

File: src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs [hunk 5]
```diff
@@ -899,19 +1045,93 @@ public void Should_Throw_If_FilePath_Is_Null()
...
+                    var result = filePath.DeserializeToIssues();
+
+                    // Then
+                    result.Count().ShouldBe(2);
+                    result.First().MessageMarkdown.ShouldBe(messageMarkdown1);
+                    result.Last().MessageMarkdown.ShouldBe(messageMarkdown2);
+                }
+                finally
+                {
+                    System.IO.File.Delete(filePath.FullPath);
```
[the remark this answers is not in the document]
Comment (pascalberger) [reply src/Cake.Issues.Tests/Serialization/IssueSerializationExtensionsTests.cs]: Good point. I'll do this in a subsequent PR over the whole codebase.

**Pull request merged**